drivers/pwm/pwm_capture.c

```diff
@@ -4,10 +4,9 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-
-#include <zephyr/zephyr.h>
 #include <zephyr/drivers/pwm.h>
 #include <zephyr/logging/log.h>
+#include <zephyr/zephyr.h>
```

Comment:
it can go wrong if headers are not designed correctly (good rules can be found e.g. here https://google.github.io/styleguide/cppguide.html). In Zephyr there are a few cases where things need to be fixed, but from my experience most files will just work even if by chance.

Reply:
Looks like this is following some 'intelligent design' after all, so I might be wrong about this, but this is the stuff that might cause issues if not done correctly and left to a regex, so while interesting and nice and I agree in general with the cleanup, it should not be part of the linter and reformatting, maybe done as part of a targeted header cleanup instead.

Reply:
While I like alphabetical headers, I don't think we want to enable that option just yet.  That is likely going to require a lot more scrutiny on the headers with tooling such as IWYU to clean them up.

Reply:
it is not alphabetical after all, it is following a more deliberare ordering, look at the .clang-config.